its/src/test/java/com/sonar/it/csharp/UnitTestProjectTypeProbingTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SonarSource :: C# :: ITs :: Plugin
+ * Copyright (C) 2011-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.csharp;
+
+import com.sonar.it.shared.TestUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+
+import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
+import static com.sonar.it.csharp.Tests.getIssues;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UnitTestProjectTypeProbingTest {
+  private static final String PROJECT = "UTProjectProbing";
+  private static Boolean isProjectAnalyzed = false;
+
+  @Rule
+  public TemporaryFolder temp = TestUtils.createTempFolder();
+
+  @Before
+  public void init() throws IOException {
+    if (!isProjectAnalyzed) {
+      TestUtils.reset(ORCHESTRATOR);
+      Tests.analyzeProject(temp, PROJECT, null);
+
+      isProjectAnalyzed = true;
+    }
+  }
+
+  @Test
+  public void mainProject_IsIdentifiedAsMain() {
+    assertThat(getIssues("UTProjectProbing:UTProjectProbing.Main/calculator.cs")).hasSize(2);

```

Comment:
you could have two String constants in the test file, one `MAIN_RULE_ID` and one `MAIN_AND_TEST_RULE_ID` and verify that, in addition to `hasSize`, the list of issues `containsExactly` the expected issue(s). 

It will help with future maintaing of the test cases

Reply:
actually I think that if you do containsExactly, it will include the size diff - and the error message will be better (as it will show the contents of the actual array)

Reply:
Indeed. I've change the code as suggested, with checking for exact occurrences of rule IDs. I think it's a nice idea.